source/common/http/conn_pool_grid.h

```diff
@@ -221,9 +222,11 @@ class ConnectivityGrid : public ConnectionPool::Instance,
   // Tracks the callbacks to be called on drain completion.
   std::list<Instance::IdleCb> idle_callbacks_;
 
-  // The connection pools to use to create new streams, ordered in the order of
-  // desired use.
-  std::list<ConnectionPool::InstancePtr> pools_;
+  // The connection pools to use to create new streams
+  ConnectionPool::InstancePtr http3_pool_;
+  ConnectionPool::InstancePtr http2_pool_;
+  // A convenience list to allow taking actions on all pools.
+  std::list<ConnectionPool::Instance*> pools_;
```

Comment:
A list involves heap allocations and pointer chasing. What would you think of absl::InlinedVector instead?